### PR TITLE
[BEAM-5100][SQL]Test Arithmetic functions at the DSL level

### PR DESCRIPTION
**Pull request by amaliujia**
1. Test math functions at the DSL level.
2. Remove `BeamSqlMathFunctionsIntegrationTest.java` as tests in it will be duplicates.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by amaliujia** [comment]
R: @akedin @apilloud

**Comment by amaliujia** [comment]
run java precommit

File: sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslSqlStdOperatorsTest.java [hunk 1]
```diff
@@ -557,6 +566,326 @@ public void testStringFunctions() throws Exception {
...
+            .addExpr("SQRT(c_smallint)", Math.sqrt(SHORT_VALUE))
+            .addExpr("SQRT(c_tinyint)", Math.sqrt(BYTE_VALUE))
+            .addExpr("SQRT(c_double)", Math.sqrt(DOUBLE_VALUE))
+            .addExpr("SQRT(c_float)", Math.sqrt(FLOAT_VALUE))
+            .addExpr("SQRT(c_decimal)", Math.sqrt(ONE.doubleValue()));
+    checker.buildRunAndCheck();
+  }
+
+  @Test
+  @SqlOperatorTest(name = "ROUND", kind = "OTHER_FUNCTION")
```
Comment (akedin) [inline sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslSqlStdOperatorsTest.java]: Can we have more granular `kind`? E.g. `"BASIC_MATH"` for things like `max, min, sqrt..`, `"TRIGONOMETRY"` for `sin, cos..`, `"RANDOM"` for randoms

**Comment by akedin** [comment]
LGTM

File: sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslSqlStdOperatorsTest.java (hunk 1 above)
Comment (amaliujia) [reply sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslSqlStdOperatorsTest.java]: `kind` comes from Calcite. It is not defined by us. See: `https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/sql/SqlKind.java`

Comment (akedin) [reply sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslSqlStdOperatorsTest.java]: Ah, didn't realize it

**Pull request merged**